src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/BreakpointEvidence.java

```diff
@@ -244,18 +245,21 @@ protected ReadEvidence( final GATKRead read, final ReadMetadata metadata, final
             this.forwardStrand = forwardStrand;
             this.cigarString = read.getCigar().toString();
             this.mappingQuality = read.getMappingQuality();
+            this.templateSize = read.getFragmentLength();
```

Comment:
My only thought about promoting template size to `ReadEvidence` is that for evidence types other than `WeirdTemplateSize`, the template length will probably not really follow a predictable distribution, and therefore might not be too useful to a machine learning module (or might cause it to overfit noise). Although, on the other hand, maybe excessively long template sizes could be downweighted or something... Also, what does the template size look like for inter-contig pairs? I guess it's worth trying out as a feature, but adding it here will create a bit more overhead for the system, ie. more information will have to be tracked in the evidence gathering stage. One other thing to note -- you should be able to get close to the template size by checking the distance between the evidence interval and the distal target interval.

Reply:
xgboost should be smart enough to ignore uninformative (e.g. constant or random) columns. Of course we don't want to throw stuff at it that we know is terrible, but so far, templateSize looks like it helps most of the time. In the more complex classifier type I presented, I can drop it or tell the classifier to ignore it on a per-evidenceType basis if cross-validation shows it's not helping.

The templateSize for interContigPairs is always 0.

Regarding performance: would it be smarter to just store the read object and add property getters? Then quantities that are kept principally for evidence-dumping like templateSize and cigarString would not be evaluated unless needed. I don't have a good sense of whether this would be an improvement, neutral, or make things worse.

Reply:
I think the read object would be quite a bit larger than the fields we're storing, as it contains the read sequence and base qualities as well as other tags. We've been trying to keep the evidence objects small, which is the only reason I asked if it was necessary to add more data to it.

Regarding learning from the data, one more thing to keep in mind: If the ML module decides, for example, to ignore evidence with really large template sizes because our benchmark data doesn't contain any true events of that size, what happens when we run on a sample that has an event (probably a very rare one) that is really huge? We don't want to miss that. Obviously this is a balancing act, and we could think of our evidence weighting as having a low prior on really huge events,  but if there are ways to engineer our features such that we can still overcome our priors in the face of good evidence that would be good. In this case I wonder if capping the template size feature might help? Just something to think about.